algorithms/kernel/covariance/covariance_helper.h

```diff
@@ -0,0 +1,64 @@
+/* file: covariance_helper.h */
+/*******************************************************************************
+* Copyright 2014-2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#ifndef __COVARIANCE_HELPER_H__
+#define __COVARIANCE_HELPER_H__
+
+#include "service/kernel/service_environment.h"
+
+namespace daal
+{
+namespace algorithms
+{
+namespace covariance
+{
+namespace internal
+{
+template <typename algorithmFPType, CpuType cpu>
+struct BSHelper
+{
+    static size_t getBlockSize(const size_t n, const size_t k)
```

Comment:
I don't see a difference btw helper for covariance/pca transform/k-means.
Can we extract this to a common class?

Reply:
Block size calculations between covariance and kmeans are different: in kmeans we split left matrix by rows, in covariance we split right matrix by columns. That has impact to algorithm block size calculation. Also we have different conditional extra memory usage in iteration of threader_for for each of GEMM based algorithms. For example that is case of pca transform when flag "isNormalized" is used. In future we have to look at for delivering the best optimizations. So, I think that we can't create general strategy for block size calculation across all algorithms. We don't have a lot of GEMM base algorithms so I think that library size won't be greatly increased.

Reply:
The difference between algorithms will be in approach of calculation of `rowsFitL1` and `rowsFitL2`.
So, we can extract part that exclude variables mentioned above.
This can be noted as future improvements, I think at the moment we can do without it.